src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
@daxian-dbw Thanks for summing-up. Await @lzybkr.
And my question remains whether to announce this cmdlet depredicated and design a new cmdlet with simple clear logic and easy to support code?